### DISCUSSION TITLE
Aggressive Monitoring Queries in Flex PSQL

### DISCUSSION THREAD

**Discussion by mswanson-mastery** [apache/airflow#36606] [Q&A]
Hello!

My DBA team has called out some rather persistant queries running on our flex server that are causing some higher-than-usual CPU / IO usage. They look to me like montioring / status queries from the scheduler but I'm not sure if there's anything that can be done about that. I'm hoping someone could point me in the right direction of the source of the queries, or maybe help me out a bit with the terminonolgy or names of this process that I could use to work backwards?

Any help / input would be much appreciated, here's the query from our server (with the DAG name cleared):

```
SELECT 

dag_run.state AS dag_run_state,
dag_run.id AS dag_run_id,
dag_run.dag_id AS dag_run_dag_id,
dag_run.queued_at AS dag_run_queued_at,
dag_run.execution_date AS dag_run_execution_date,
dag_run.start_date AS dag_run_start_date,
dag_run.end_date AS dag_run_end_date,
dag_run.run_id AS dag_run_run_id,
dag_run.creating_job_id AS dag_run_creating_job_id,
dag_run.external_trigger AS dag_run_external_trigger,
dag_run.run_type AS dag_run_run_type,
dag_run.conf AS dag_run_conf,
dag_run.data_interval_start AS dag_run_data_interval_start,
dag_run.data_interval_end AS dag_run_data_interval_end,
dag_run.last_scheduling_decision AS dag_run_last_scheduling_decision,
dag_run.dag_hash AS dag_run_dag_hash,
dag_run.log_template_id AS dag_run_log_template_id,
dag_run.updated_at AS dag_run_updated_at  

FROM dag_run  
WHERE dag_run.dag_id = '<DAG NAME>' 
AND dag_run.run_id = 'scheduled__2023-11-02T03:55:00+00:00' 
FOR UPDATE
;
```

Thanks!

**Comment by potiuk** [comment]
Airflow's scheduler uses those queries to schedule stuff. Yes. They are running continously - Airflow runs them to check what should be scheduled next - depending on number of dags, tasks, and a number of other factors it might be more or less demanding on the DB, but it's generally expected to be relatively heavy.

If you want to take a look at optimizing it here is the whole document that describes how you can approach fine-tuninng scheduler performance - explains all the knobs you can turn and what effect they might have. If you want to optimize the way your DB is used, you likely need to get deeply familiar with it and likely perform some experiments based on the explanations of the knobs you can turn https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/scheduler.html#fine-tuning-your-scheduler-performance

**Comment by mswanson-mastery** [comment]
@potiuk 
Thank you so much Jarek - this is exactly what I needed! I really appreciate the thoughtful answer and the doc link, I'll dig in from there and bring the recommendations back to the team.
